{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# featureCounts Compare Exon-Intron Boundaries for non DMS samples: Madrasin Vs Controls\n",
    "\n",
    "## Purpose\n",
    "\n",
    "This were read counts generated by the program featureCounts After STAR alignment\n",
    "\n",
    "Aim of this script is to get the ratio of Madrasin vs Controls in exon-intron boundaries for samples that were not treated with DMS \n",
    "\n",
    "We want to see how the ratio changes as you add varying concentrations of Madrasin"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Import packages\n",
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "import os\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Set variable to find highly expressed genes or transcripts\n",
    "feature=\"Gene\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Set variable to remove non-coding genes\n",
    "removeNC=False\n",
    "removeNC_string=\"\"\n",
    "if removeNC:\n",
    "    removeNC_string=\"Coding\"\n",
    "    geneNamesAndTranscriptIDs = pd.read_csv(\"../data/GeneNameTranscriptID_NCBI_RefSeq_hg38_FromGFFfile.tsv\",header=None,sep=\"\\t\")\n",
    "    noncodinggeneNames= list(geneNamesAndTranscriptIDs[geneNamesAndTranscriptIDs[0].str.contains(\"NR_\")|geneNamesAndTranscriptIDs[0].str.contains(\"XR_\")][1].values)\n",
    "    noncodingtranscriptNames= list(geneNamesAndTranscriptIDs[geneNamesAndTranscriptIDs[0].str.contains(\"NR_\")|geneNamesAndTranscriptIDs[0].str.contains(\"XR_\")][0].values)\n",
    "    noncodingFeatures = noncodinggeneNames+noncodingtranscriptNames"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(153118, 2)\n"
     ]
    }
   ],
   "source": [
    "# Set variable to only look at features that have introns in them\n",
    "featuresWithIntrons=True\n",
    "featuresWithIntrons_string=\"\"\n",
    "if featuresWithIntrons:\n",
    "    featuresWithIntrons_string=\"WithIntronicRegions\"\n",
    "    # Get the transcript and genes that have introns within them\n",
    "    geneNamesAndTranscriptIDs_withIntrons = pd.read_csv(\"../data/GeneNameTranscriptID_NCBI_RefSeq_hg38_FromGFFfile_OnlyOnesWithIntrons.tsv\",header=None,sep=\"\\t\")\n",
    "    print geneNamesAndTranscriptIDs_withIntrons.shape\n",
    "    geneNamesAndTranscriptIDs_withIntrons.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Set variable to determine how many expressed features to look at\n",
    "# If set to 0, then get all features\n",
    "numFeatures=50"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# which end\n",
    "whichend=str(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Get highly expressed features from each file for the transcriptome data\n",
    "top_features_Transcriptome = []\n",
    "for filename in os.listdir(\"../tmp/Lela_MadrasinReadsTest_June2018/\"):\n",
    "    if (\"ExonIntron\"+whichend+\"p_featureCounts_JustCounts.txt\" in filename) and ((\"minus\" in filename and \"Mad-30mm-8hr-minus\" not in filename) or (\"Mad-30mm-8hr-DMS\" in filename) or (\"All_Minus\" in filename)):\n",
    "        print filename\n",
    "        # Read the salmon output file\n",
    "        with open(\"../tmp/Lela_MadrasinReadsTest_June2018/\"+filename) as f:\n",
    "            lines = f.readlines()\n",
    "        with open(\"../tmp/Lela_MadrasinReadsTest_June2018/\"+filename.split('.')[0]+\"_DropHeader.txt\",'w') as fw:\n",
    "            fw.writelines(lines[1:])\n",
    "        data_to_read = pd.read_csv(\"../tmp/Lela_MadrasinReadsTest_June2018/\"+filename.split('.')[0]+\"_DropHeader.txt\",header=0,sep=\"\\t\")\n",
    "        data_to_read.columns = [\"Name\",\"Length\",\"Count\"]\n",
    "        # If features with Introns flag is set, only get features that have introns in them\n",
    "        if featuresWithIntrons:\n",
    "            # Only features with introns\n",
    "            data_to_read = data_to_read[(data_to_read[\"Name\"].isin(geneNamesAndTranscriptIDs_withIntrons[0])|data_to_read[\"Name\"].isin(geneNamesAndTranscriptIDs_withIntrons[1]))]\n",
    "        # If removeNC flag is set, only get coding features\n",
    "        if removeNC:\n",
    "            data_to_read = data_to_read[~(data_to_read[\"Name\"].str.contains(\"NR_\")|data_to_read[\"Name\"].str.contains(\"XR_\")|data_to_read[\"Name\"].isin(noncodingFeatures))]\n",
    "        # Calculate TPMs\n",
    "        RPKs = 1000*(data_to_read[\"Count\"]/data_to_read[\"Length\"])\n",
    "        total_RPKs = RPKs.sum()\n",
    "        TPMs = 1000000*(RPKs/total_RPKs)\n",
    "        data_to_read = data_to_read.assign(TPM=TPMs)\n",
    "        # Sort the data by TPMs\n",
    "        data_to_read_sorted = data_to_read.sort_values(by=[\"TPM\"],ascending=False)\n",
    "        # Grab the names of the first numFeatures features\n",
    "        if numFeatures==0:\n",
    "            top_features_Transcriptome.extend(list(data_to_read_sorted[\"Name\"].values))\n",
    "        else:\n",
    "            top_features_Transcriptome.extend(list(data_to_read_sorted.iloc[range(numFeatures),0].values))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Combine the two feature files together to get a unique list\n",
    "unique_top_features = list(set(top_features_Transcriptome))\n",
    "print(len(unique_top_features))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Create a data frame with unique top features as column\n",
    "feature_readCounts_transcriptome = pd.DataFrame({\"feature\":sorted(unique_top_features)})\n",
    "# For the unique features, get the TPM for each of them and store in a data frame\n",
    "# Read in the Count files \n",
    "for filename in os.listdir(\"../tmp/Lela_MadrasinReadsTest_June2018/\"):\n",
    "    current_columns = list(feature_readCounts_transcriptome.columns.values)\n",
    "    if (\"ExonIntron\"+whichend+\"p_featureCounts_JustCounts_DropHeader.txt\" in filename) and ((\"minus\" in filename and \"Mad-30mm-8hr-minus\" not in filename) or (\"Mad-30mm-8hr-DMS\" in filename) or (\"All_Minus\" in filename)):\n",
    "        print filename\n",
    "        current_columns.append(filename.split(\"_ExonIntron\")[0])\n",
    "        data_to_read = pd.read_csv(\"../tmp/Lela_MadrasinReadsTest_June2018/\"+filename,header=0,sep=\"\\t\")\n",
    "        data_to_read.columns = [\"Name\",\"Length\",\"Count\"]\n",
    "        # Calculate TPMs\n",
    "        RPKs = 1000*(data_to_read[\"Count\"]/data_to_read[\"Length\"])\n",
    "        total_RPKs = RPKs.sum()\n",
    "        TPMs = 1000000*(RPKs/total_RPKs)\n",
    "        data_to_read = data_to_read.assign(TPM=TPMs)\n",
    "        data_GetFeatures = data_to_read[data_to_read[\"Name\"].isin(unique_top_features)]\n",
    "        data_sortedBy_FeatureName = data_GetFeatures.sort_values(by=\"Name\")\n",
    "        data_sortedBy_FeatureName = data_sortedBy_FeatureName.reset_index(drop=True)\n",
    "        feature_readCounts_transcriptome = pd.concat([feature_readCounts_transcriptome,data_sortedBy_FeatureName[\"TPM\"]],axis=1,ignore_index=True)\n",
    "        feature_readCounts_transcriptome.columns = current_columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "feature_readCounts_transcriptome = feature_readCounts_transcriptome.iloc[:,range(1,feature_readCounts_transcriptome.shape[1])]\n",
    "feature_readCounts_transcriptome.index = sorted(unique_top_features)\n",
    "print feature_readCounts_transcriptome.shape\n",
    "feature_readCounts_transcriptome.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "sorted_columns = sorted(list(feature_readCounts_transcriptome.columns.values))\n",
    "sorted_columns_NoControl = [i for i in sorted_columns if \"All\" not in i]"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.15rc1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
