{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# featureCounts Compare Exon-Intron Boundaries for Madrasin: DMS vs non-DMS\n",
    "\n",
    "DataSet: Madrasin HiSeq Data from Oct 2018\n",
    "\n",
    "## Purpose\n",
    "\n",
    "This were read counts generated by the program featureCounts After Alignment using Bowtie2 ShapeMapper2 Parms\n",
    "\n",
    "Aim of this script is to get the ratio of Madrasin vs Controls in exon-intron boundaries for samples that were not treated with DMS \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Import packages\n",
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "import os\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Set variable to find highly expressed genes or transcripts\n",
    "feature=\"Gene\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Set variable to remove non-coding genes\n",
    "removeNC=False\n",
    "removeNC_string=\"\"\n",
    "if removeNC:\n",
    "    removeNC_string=\"Coding\"\n",
    "    geneNamesAndTranscriptIDs = pd.read_csv(\"../data/GeneNameTranscriptID_NCBI_RefSeq_hg38_FromGFFfile.tsv\",header=None,sep=\"\\t\")\n",
    "    noncodinggeneNames= list(geneNamesAndTranscriptIDs[geneNamesAndTranscriptIDs[0].str.contains(\"NR_\")|geneNamesAndTranscriptIDs[0].str.contains(\"XR_\")][1].values)\n",
    "    noncodingtranscriptNames= list(geneNamesAndTranscriptIDs[geneNamesAndTranscriptIDs[0].str.contains(\"NR_\")|geneNamesAndTranscriptIDs[0].str.contains(\"XR_\")][0].values)\n",
    "    noncodingFeatures = noncodinggeneNames+noncodingtranscriptNames"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(153118, 2)\n"
     ]
    }
   ],
   "source": [
    "# Set variable to only look at features that have introns in them\n",
    "featuresWithIntrons=True\n",
    "featuresWithIntrons_string=\"\"\n",
    "if featuresWithIntrons:\n",
    "    featuresWithIntrons_string=\"WithIntronicRegions\"\n",
    "    # Get the transcript and genes that have introns within them\n",
    "    geneNamesAndTranscriptIDs_withIntrons = pd.read_csv(\"../data/GeneNameTranscriptID_NCBI_RefSeq_hg38_FromGFFfile_OnlyOnesWithIntrons.tsv\",header=None,sep=\"\\t\")\n",
    "    print geneNamesAndTranscriptIDs_withIntrons.shape\n",
    "    geneNamesAndTranscriptIDs_withIntrons.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Set variable to determine how many expressed features to look at\n",
    "# If set to 0, then get all features\n",
    "numFeatures=50"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "# which end\n",
    "whichend=str(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "NoDMSSample_CTCTCTAC_S2_ExonIntron5p_featureCounts_JustCounts.txt\n",
      "DMSTreatedSample_TAGGCATG_S1_ExonIntron5p_featureCounts_JustCounts.txt\n"
     ]
    }
   ],
   "source": [
    "# Get highly expressed features from each file for the transcriptome data\n",
    "top_features_Transcriptome = []\n",
    "for filename in os.listdir(\"../tmp/Madrasin_HiSeq_October2018_alignments/\"):\n",
    "    if (\"_Intron\"+whichend+\"p_featureCounts_JustCounts.txt\" in filename):\n",
    "        print filename\n",
    "        # Read the salmon output file\n",
    "        with open(\"../tmp/Madrasin_HiSeq_October2018_alignments/\"+filename) as f:\n",
    "            lines = f.readlines()\n",
    "        with open(\"../tmp/Madrasin_HiSeq_October2018_alignments/\"+filename.split('.')[0]+\"_DropHeader.txt\",'w') as fw:\n",
    "            fw.writelines(lines[1:])\n",
    "        data_to_read = pd.read_csv(\"../tmp/Madrasin_HiSeq_October2018_alignments/\"+filename.split('.')[0]+\"_DropHeader.txt\",header=0,sep=\"\\t\")\n",
    "        data_to_read.columns = [\"Name\",\"Length\",\"Count\"]\n",
    "        # If features with Introns flag is set, only get features that have introns in them\n",
    "        if featuresWithIntrons:\n",
    "            # Only features with introns\n",
    "            data_to_read = data_to_read[(data_to_read[\"Name\"].isin(geneNamesAndTranscriptIDs_withIntrons[0])|data_to_read[\"Name\"].isin(geneNamesAndTranscriptIDs_withIntrons[1]))]\n",
    "        # If removeNC flag is set, only get coding features\n",
    "        if removeNC:\n",
    "            data_to_read = data_to_read[~(data_to_read[\"Name\"].str.contains(\"NR_\")|data_to_read[\"Name\"].str.contains(\"XR_\")|data_to_read[\"Name\"].isin(noncodingFeatures))]\n",
    "        # Calculate TPMs\n",
    "        RPKs = 1000*(data_to_read[\"Count\"]/data_to_read[\"Length\"])\n",
    "        total_RPKs = RPKs.sum()\n",
    "        TPMs = 1000000*(RPKs/total_RPKs)\n",
    "        data_to_read = data_to_read.assign(TPM=TPMs)\n",
    "        # Sort the data by TPMs\n",
    "        data_to_read_sorted = data_to_read.sort_values(by=[\"TPM\"],ascending=False)\n",
    "        # Grab the names of the first numFeatures features\n",
    "        if numFeatures==0:\n",
    "            top_features_Transcriptome.extend(list(data_to_read_sorted[\"Name\"].values))\n",
    "        else:\n",
    "            top_features_Transcriptome.extend(list(data_to_read_sorted.iloc[range(numFeatures),0].values))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "72\n"
     ]
    }
   ],
   "source": [
    "# Combine the two feature files together to get a unique list\n",
    "unique_top_features = list(set(top_features_Transcriptome))\n",
    "print(len(unique_top_features))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "DMSTreatedSample_TAGGCATG_S1_ExonIntron5p_featureCounts_JustCounts_DropHeader.txt\n",
      "NoDMSSample_CTCTCTAC_S2_ExonIntron5p_featureCounts_JustCounts_DropHeader.txt\n"
     ]
    }
   ],
   "source": [
    "# Create a data frame with unique top features as column\n",
    "feature_readCounts_transcriptome = pd.DataFrame({\"feature\":sorted(unique_top_features)})\n",
    "# For the unique features, get the TPM for each of them and store in a data frame\n",
    "# Read in the Count files \n",
    "for filename in os.listdir(\"../tmp/Madrasin_HiSeq_October2018_alignments/\"):\n",
    "    current_columns = list(feature_readCounts_transcriptome.columns.values)\n",
    "    if (\"ExonIntron\"+whichend+\"p_featureCounts_JustCounts_DropHeader.txt\" in filename):\n",
    "        print filename\n",
    "        current_columns.append(filename.split(\"_ExonIntron\")[0])\n",
    "        data_to_read = pd.read_csv(\"../tmp/Madrasin_HiSeq_October2018_alignments/\"+filename,header=0,sep=\"\\t\")\n",
    "        data_to_read.columns = [\"Name\",\"Length\",\"Count\"]\n",
    "        # Calculate TPMs\n",
    "        RPKs = 1000*(data_to_read[\"Count\"]/data_to_read[\"Length\"])\n",
    "        total_RPKs = RPKs.sum()\n",
    "        TPMs = 1000000*(RPKs/total_RPKs)\n",
    "        data_to_read = data_to_read.assign(TPM=TPMs)\n",
    "        data_GetFeatures = data_to_read[data_to_read[\"Name\"].isin(unique_top_features)]\n",
    "        data_sortedBy_FeatureName = data_GetFeatures.sort_values(by=\"Name\")\n",
    "        data_sortedBy_FeatureName = data_sortedBy_FeatureName.reset_index(drop=True)\n",
    "        feature_readCounts_transcriptome = pd.concat([feature_readCounts_transcriptome,data_sortedBy_FeatureName[\"TPM\"]],axis=1,ignore_index=True)\n",
    "        feature_readCounts_transcriptome.columns = current_columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(72, 2)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>DMSTreatedSample_TAGGCATG_S1</th>\n",
       "      <th>NoDMSSample_CTCTCTAC_S2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ACTB</th>\n",
       "      <td>8648.644113</td>\n",
       "      <td>7566.765250</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ACTG1</th>\n",
       "      <td>2431.150595</td>\n",
       "      <td>1786.700069</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AEN</th>\n",
       "      <td>837.012411</td>\n",
       "      <td>531.387287</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ATF4</th>\n",
       "      <td>819.664307</td>\n",
       "      <td>759.797528</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ATP5F1B</th>\n",
       "      <td>802.709685</td>\n",
       "      <td>2295.300607</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         DMSTreatedSample_TAGGCATG_S1  NoDMSSample_CTCTCTAC_S2\n",
       "ACTB                      8648.644113              7566.765250\n",
       "ACTG1                     2431.150595              1786.700069\n",
       "AEN                        837.012411               531.387287\n",
       "ATF4                       819.664307               759.797528\n",
       "ATP5F1B                    802.709685              2295.300607"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "feature_readCounts_transcriptome = feature_readCounts_transcriptome.iloc[:,range(1,feature_readCounts_transcriptome.shape[1])]\n",
    "feature_readCounts_transcriptome.index = sorted(unique_top_features)\n",
    "print feature_readCounts_transcriptome.shape\n",
    "feature_readCounts_transcriptome.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>DMSTreatedSample_TAGGCATG_S1</th>\n",
       "      <th>NoDMSSample_CTCTCTAC_S2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ACTB</th>\n",
       "      <td>8648.644113</td>\n",
       "      <td>7566.765250</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ACTG1</th>\n",
       "      <td>2431.150595</td>\n",
       "      <td>1786.700069</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AEN</th>\n",
       "      <td>837.012411</td>\n",
       "      <td>531.387287</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ATF4</th>\n",
       "      <td>819.664307</td>\n",
       "      <td>759.797528</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ATP5F1B</th>\n",
       "      <td>802.709685</td>\n",
       "      <td>2295.300607</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CALR</th>\n",
       "      <td>1294.579868</td>\n",
       "      <td>3068.165441</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CCND2</th>\n",
       "      <td>828.943525</td>\n",
       "      <td>968.306271</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CD48</th>\n",
       "      <td>671.196810</td>\n",
       "      <td>1546.487191</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CD70</th>\n",
       "      <td>1489.142742</td>\n",
       "      <td>882.516570</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CD74</th>\n",
       "      <td>3907.453162</td>\n",
       "      <td>3916.404007</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CD79A</th>\n",
       "      <td>570.011547</td>\n",
       "      <td>1051.739228</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CDKN1A</th>\n",
       "      <td>1207.865237</td>\n",
       "      <td>580.011096</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CFL1</th>\n",
       "      <td>910.738118</td>\n",
       "      <td>1486.263958</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CORO1A</th>\n",
       "      <td>2160.710759</td>\n",
       "      <td>3577.888705</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DUSP2</th>\n",
       "      <td>872.246545</td>\n",
       "      <td>710.305898</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EEF1A1</th>\n",
       "      <td>1856.580159</td>\n",
       "      <td>4013.691854</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EEF2</th>\n",
       "      <td>2309.494398</td>\n",
       "      <td>3139.778603</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EIF2S3B</th>\n",
       "      <td>415.502787</td>\n",
       "      <td>1256.733897</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EIF4A2</th>\n",
       "      <td>734.707906</td>\n",
       "      <td>1049.020555</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FLNA</th>\n",
       "      <td>726.353694</td>\n",
       "      <td>1029.839131</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GADD45B</th>\n",
       "      <td>1250.288783</td>\n",
       "      <td>761.817594</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GADD45GIP1</th>\n",
       "      <td>898.022152</td>\n",
       "      <td>735.556729</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GAPDH</th>\n",
       "      <td>1749.340665</td>\n",
       "      <td>3003.650307</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GRN</th>\n",
       "      <td>873.889881</td>\n",
       "      <td>792.618371</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSTP1</th>\n",
       "      <td>819.320632</td>\n",
       "      <td>949.305020</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HIST1H3D</th>\n",
       "      <td>1500.140335</td>\n",
       "      <td>2508.922644</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HLA-A</th>\n",
       "      <td>539.466470</td>\n",
       "      <td>1021.009605</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HLA-B</th>\n",
       "      <td>1024.446198</td>\n",
       "      <td>1366.332162</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HLA-DQA1</th>\n",
       "      <td>952.150927</td>\n",
       "      <td>2133.947792</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HLA-DQB1</th>\n",
       "      <td>670.578195</td>\n",
       "      <td>999.175413</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OAZ1</th>\n",
       "      <td>819.922063</td>\n",
       "      <td>1179.845114</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PCNA</th>\n",
       "      <td>635.060471</td>\n",
       "      <td>1024.632547</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PCNP</th>\n",
       "      <td>737.412955</td>\n",
       "      <td>1502.903301</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PFN1</th>\n",
       "      <td>2091.948274</td>\n",
       "      <td>2781.252865</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PPIB</th>\n",
       "      <td>636.915252</td>\n",
       "      <td>1184.200883</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RACK1</th>\n",
       "      <td>1175.220396</td>\n",
       "      <td>1471.329894</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RBM14</th>\n",
       "      <td>1210.422511</td>\n",
       "      <td>626.978153</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RBM4</th>\n",
       "      <td>783.578457</td>\n",
       "      <td>497.693894</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RPL10</th>\n",
       "      <td>1489.935884</td>\n",
       "      <td>1210.420797</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RPL13A</th>\n",
       "      <td>1008.538206</td>\n",
       "      <td>520.292943</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RPL15</th>\n",
       "      <td>489.443707</td>\n",
       "      <td>996.776857</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RPL18</th>\n",
       "      <td>1450.994844</td>\n",
       "      <td>1126.439605</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RPL27A</th>\n",
       "      <td>940.036392</td>\n",
       "      <td>629.124473</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RPL3</th>\n",
       "      <td>1282.688962</td>\n",
       "      <td>854.282836</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RPL8</th>\n",
       "      <td>784.956513</td>\n",
       "      <td>1321.952355</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RPS11</th>\n",
       "      <td>1061.955014</td>\n",
       "      <td>889.902438</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RPS19</th>\n",
       "      <td>1247.651508</td>\n",
       "      <td>600.459618</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RPS4X</th>\n",
       "      <td>527.368922</td>\n",
       "      <td>1098.411182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RPS5</th>\n",
       "      <td>872.452750</td>\n",
       "      <td>983.115884</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RPS6</th>\n",
       "      <td>1041.953142</td>\n",
       "      <td>1648.980318</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RPS8</th>\n",
       "      <td>1431.886527</td>\n",
       "      <td>1262.945602</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SIRT4</th>\n",
       "      <td>2433.732825</td>\n",
       "      <td>6043.029058</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SNHG10</th>\n",
       "      <td>798.012797</td>\n",
       "      <td>681.772458</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SRSF1</th>\n",
       "      <td>461.383368</td>\n",
       "      <td>1057.126072</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SRSF2</th>\n",
       "      <td>628.667058</td>\n",
       "      <td>1544.782760</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TRAF4</th>\n",
       "      <td>893.325264</td>\n",
       "      <td>716.534436</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TRIM28</th>\n",
       "      <td>1014.184556</td>\n",
       "      <td>582.066832</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TRIR</th>\n",
       "      <td>798.223853</td>\n",
       "      <td>753.430753</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TUBA1B</th>\n",
       "      <td>777.735986</td>\n",
       "      <td>1671.857571</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VIM</th>\n",
       "      <td>577.717274</td>\n",
       "      <td>1225.170357</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>72 rows Ã— 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            DMSTreatedSample_TAGGCATG_S1  NoDMSSample_CTCTCTAC_S2\n",
       "ACTB                         8648.644113              7566.765250\n",
       "ACTG1                        2431.150595              1786.700069\n",
       "AEN                           837.012411               531.387287\n",
       "ATF4                          819.664307               759.797528\n",
       "ATP5F1B                       802.709685              2295.300607\n",
       "CALR                         1294.579868              3068.165441\n",
       "CCND2                         828.943525               968.306271\n",
       "CD48                          671.196810              1546.487191\n",
       "CD70                         1489.142742               882.516570\n",
       "CD74                         3907.453162              3916.404007\n",
       "CD79A                         570.011547              1051.739228\n",
       "CDKN1A                       1207.865237               580.011096\n",
       "CFL1                          910.738118              1486.263958\n",
       "CORO1A                       2160.710759              3577.888705\n",
       "DUSP2                         872.246545               710.305898\n",
       "EEF1A1                       1856.580159              4013.691854\n",
       "EEF2                         2309.494398              3139.778603\n",
       "EIF2S3B                       415.502787              1256.733897\n",
       "EIF4A2                        734.707906              1049.020555\n",
       "FLNA                          726.353694              1029.839131\n",
       "GADD45B                      1250.288783               761.817594\n",
       "GADD45GIP1                    898.022152               735.556729\n",
       "GAPDH                        1749.340665              3003.650307\n",
       "GRN                           873.889881               792.618371\n",
       "GSTP1                         819.320632               949.305020\n",
       "HIST1H3D                     1500.140335              2508.922644\n",
       "HLA-A                         539.466470              1021.009605\n",
       "HLA-B                        1024.446198              1366.332162\n",
       "HLA-DQA1                      952.150927              2133.947792\n",
       "HLA-DQB1                      670.578195               999.175413\n",
       "...                                  ...                      ...\n",
       "OAZ1                          819.922063              1179.845114\n",
       "PCNA                          635.060471              1024.632547\n",
       "PCNP                          737.412955              1502.903301\n",
       "PFN1                         2091.948274              2781.252865\n",
       "PPIB                          636.915252              1184.200883\n",
       "RACK1                        1175.220396              1471.329894\n",
       "RBM14                        1210.422511               626.978153\n",
       "RBM4                          783.578457               497.693894\n",
       "RPL10                        1489.935884              1210.420797\n",
       "RPL13A                       1008.538206               520.292943\n",
       "RPL15                         489.443707               996.776857\n",
       "RPL18                        1450.994844              1126.439605\n",
       "RPL27A                        940.036392               629.124473\n",
       "RPL3                         1282.688962               854.282836\n",
       "RPL8                          784.956513              1321.952355\n",
       "RPS11                        1061.955014               889.902438\n",
       "RPS19                        1247.651508               600.459618\n",
       "RPS4X                         527.368922              1098.411182\n",
       "RPS5                          872.452750               983.115884\n",
       "RPS6                         1041.953142              1648.980318\n",
       "RPS8                         1431.886527              1262.945602\n",
       "SIRT4                        2433.732825              6043.029058\n",
       "SNHG10                        798.012797               681.772458\n",
       "SRSF1                         461.383368              1057.126072\n",
       "SRSF2                         628.667058              1544.782760\n",
       "TRAF4                         893.325264               716.534436\n",
       "TRIM28                       1014.184556               582.066832\n",
       "TRIR                          798.223853               753.430753\n",
       "TUBA1B                        777.735986              1671.857571\n",
       "VIM                           577.717274              1225.170357\n",
       "\n",
       "[72 rows x 2 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "feature_readCounts_transcriptome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.15rc1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
