{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Check for "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>phenotype</th>\n",
       "      <th>event_id</th>\n",
       "      <th>modality</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>MN</td>\n",
       "      <td>exon:chr10:101165513-101165617:-@exon:chr10:10...</td>\n",
       "      <td>included</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>MN</td>\n",
       "      <td>exon:chr10:101480744-101480825:-@exon:chr10:10...</td>\n",
       "      <td>bimodal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>MN</td>\n",
       "      <td>exon:chr10:102114184-102114389:+@exon:chr10:10...</td>\n",
       "      <td>included</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>MN</td>\n",
       "      <td>exon:chr10:102286732-102286831:-@exon:chr10:10...</td>\n",
       "      <td>included</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>MN</td>\n",
       "      <td>exon:chr10:102747070-102747240:-@exon:chr10:10...</td>\n",
       "      <td>bimodal</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Unnamed: 0 phenotype                                           event_id  \\\n",
       "0           0        MN  exon:chr10:101165513-101165617:-@exon:chr10:10...   \n",
       "1           1        MN  exon:chr10:101480744-101480825:-@exon:chr10:10...   \n",
       "2           2        MN  exon:chr10:102114184-102114389:+@exon:chr10:10...   \n",
       "3           3        MN  exon:chr10:102286732-102286831:-@exon:chr10:10...   \n",
       "4           4        MN  exon:chr10:102747070-102747240:-@exon:chr10:10...   \n",
       "\n",
       "   modality  \n",
       "0  included  \n",
       "1   bimodal  \n",
       "2  included  \n",
       "3  included  \n",
       "4   bimodal  "
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "modalities = pd.read_csv(\"../data/GSE85908_modalities_tidy.csv\")\n",
    "modalities.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>phenotype</th>\n",
       "      <th>event_id</th>\n",
       "      <th>modality</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MN</td>\n",
       "      <td>exon:chr10:101165513-101165617:-@exon:chr10:10...</td>\n",
       "      <td>included</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MN</td>\n",
       "      <td>exon:chr10:101480744-101480825:-@exon:chr10:10...</td>\n",
       "      <td>bimodal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MN</td>\n",
       "      <td>exon:chr10:102114184-102114389:+@exon:chr10:10...</td>\n",
       "      <td>included</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MN</td>\n",
       "      <td>exon:chr10:102286732-102286831:-@exon:chr10:10...</td>\n",
       "      <td>included</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MN</td>\n",
       "      <td>exon:chr10:102747070-102747240:-@exon:chr10:10...</td>\n",
       "      <td>bimodal</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  phenotype                                           event_id  modality\n",
       "0        MN  exon:chr10:101165513-101165617:-@exon:chr10:10...  included\n",
       "1        MN  exon:chr10:101480744-101480825:-@exon:chr10:10...   bimodal\n",
       "2        MN  exon:chr10:102114184-102114389:+@exon:chr10:10...  included\n",
       "3        MN  exon:chr10:102286732-102286831:-@exon:chr10:10...  included\n",
       "4        MN  exon:chr10:102747070-102747240:-@exon:chr10:10...   bimodal"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Remove first column\n",
    "modalities = modalities.iloc[:,[1,2,3]]\n",
    "modalities.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(7079, 3)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>phenotype</th>\n",
       "      <th>event_id</th>\n",
       "      <th>modality</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MN</td>\n",
       "      <td>exon:chr10:101480744-101480825:-@exon:chr10:10...</td>\n",
       "      <td>bimodal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MN</td>\n",
       "      <td>exon:chr10:102747070-102747240:-@exon:chr10:10...</td>\n",
       "      <td>bimodal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>MN</td>\n",
       "      <td>exon:chr10:103113864-103113985:+@exon:chr10:10...</td>\n",
       "      <td>bimodal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>MN</td>\n",
       "      <td>exon:chr10:103360960-103361093:+@exon:chr10:10...</td>\n",
       "      <td>bimodal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>MN</td>\n",
       "      <td>exon:chr10:103360960-103361093:+@exon:chr10:10...</td>\n",
       "      <td>bimodal</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   phenotype                                           event_id modality\n",
       "1         MN  exon:chr10:101480744-101480825:-@exon:chr10:10...  bimodal\n",
       "4         MN  exon:chr10:102747070-102747240:-@exon:chr10:10...  bimodal\n",
       "6         MN  exon:chr10:103113864-103113985:+@exon:chr10:10...  bimodal\n",
       "10        MN  exon:chr10:103360960-103361093:+@exon:chr10:10...  bimodal\n",
       "12        MN  exon:chr10:103360960-103361093:+@exon:chr10:10...  bimodal"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Get all bimodal events\n",
    "bimodal_events = modalities[modalities[\"modality\"]==\"bimodal\"]\n",
    "print bimodal_events.shape\n",
    "bimodal_events.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1316, 3)\n"
     ]
    }
   ],
   "source": [
    "# Get MN bimodal events\n",
    "MN_bimodal_events = bimodal_events[bimodal_events[\"phenotype\"]==\"MN\"]\n",
    "print MN_bimodal_events.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(2325, 3)\n"
     ]
    }
   ],
   "source": [
    "# Get NPC bimodal events\n",
    "NPC_bimodal_events = bimodal_events[bimodal_events[\"phenotype\"]==\"NPC\"]\n",
    "print NPC_bimodal_events.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(3438, 3)\n"
     ]
    }
   ],
   "source": [
    "# Get iPSC bimodal events\n",
    "iPSC_bimodal_events = bimodal_events[bimodal_events[\"phenotype\"]==\"iPSC\"]\n",
    "print iPSC_bimodal_events.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>phenotype</th>\n",
       "      <th>event_id</th>\n",
       "      <th>modality</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>14146</th>\n",
       "      <td>iPSC</td>\n",
       "      <td>exon:chr10:100190328-100190427:-@exon:chr10:10...</td>\n",
       "      <td>bimodal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14147</th>\n",
       "      <td>iPSC</td>\n",
       "      <td>exon:chr10:100195392-100195529:-@exon:chr10:10...</td>\n",
       "      <td>bimodal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14150</th>\n",
       "      <td>iPSC</td>\n",
       "      <td>exon:chr10:101507014-101507147:+@exon:chr10:10...</td>\n",
       "      <td>bimodal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14151</th>\n",
       "      <td>iPSC</td>\n",
       "      <td>exon:chr10:101507014-101507147:+@exon:chr10:10...</td>\n",
       "      <td>bimodal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14154</th>\n",
       "      <td>iPSC</td>\n",
       "      <td>exon:chr10:102013178-102013296:-@exon:chr10:10...</td>\n",
       "      <td>bimodal</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      phenotype                                           event_id modality\n",
       "14146      iPSC  exon:chr10:100190328-100190427:-@exon:chr10:10...  bimodal\n",
       "14147      iPSC  exon:chr10:100195392-100195529:-@exon:chr10:10...  bimodal\n",
       "14150      iPSC  exon:chr10:101507014-101507147:+@exon:chr10:10...  bimodal\n",
       "14151      iPSC  exon:chr10:101507014-101507147:+@exon:chr10:10...  bimodal\n",
       "14154      iPSC  exon:chr10:102013178-102013296:-@exon:chr10:10...  bimodal"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "iPSC_bimodal_events.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>phenotype</th>\n",
       "      <th>event_id</th>\n",
       "      <th>modality</th>\n",
       "      <th>alt_exon_chrom</th>\n",
       "      <th>alt_exon_start</th>\n",
       "      <th>alt_exon_end</th>\n",
       "      <th>alt_exon_strand</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>14146</th>\n",
       "      <td>iPSC</td>\n",
       "      <td>exon:chr10:100190328-100190427:-@exon:chr10:10...</td>\n",
       "      <td>bimodal</td>\n",
       "      <td>chr10</td>\n",
       "      <td>100189548</td>\n",
       "      <td>100189646</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14147</th>\n",
       "      <td>iPSC</td>\n",
       "      <td>exon:chr10:100195392-100195529:-@exon:chr10:10...</td>\n",
       "      <td>bimodal</td>\n",
       "      <td>chr10</td>\n",
       "      <td>100195029</td>\n",
       "      <td>100195171</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14150</th>\n",
       "      <td>iPSC</td>\n",
       "      <td>exon:chr10:101507014-101507147:+@exon:chr10:10...</td>\n",
       "      <td>bimodal</td>\n",
       "      <td>chr10</td>\n",
       "      <td>101510126</td>\n",
       "      <td>101510153</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14151</th>\n",
       "      <td>iPSC</td>\n",
       "      <td>exon:chr10:101507014-101507147:+@exon:chr10:10...</td>\n",
       "      <td>bimodal</td>\n",
       "      <td>chr10</td>\n",
       "      <td>101514286</td>\n",
       "      <td>101514391</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14154</th>\n",
       "      <td>iPSC</td>\n",
       "      <td>exon:chr10:102013178-102013296:-@exon:chr10:10...</td>\n",
       "      <td>bimodal</td>\n",
       "      <td>chr10</td>\n",
       "      <td>102010005</td>\n",
       "      <td>102010089</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      phenotype                                           event_id modality  \\\n",
       "14146      iPSC  exon:chr10:100190328-100190427:-@exon:chr10:10...  bimodal   \n",
       "14147      iPSC  exon:chr10:100195392-100195529:-@exon:chr10:10...  bimodal   \n",
       "14150      iPSC  exon:chr10:101507014-101507147:+@exon:chr10:10...  bimodal   \n",
       "14151      iPSC  exon:chr10:101507014-101507147:+@exon:chr10:10...  bimodal   \n",
       "14154      iPSC  exon:chr10:102013178-102013296:-@exon:chr10:10...  bimodal   \n",
       "\n",
       "      alt_exon_chrom alt_exon_start alt_exon_end alt_exon_strand  \n",
       "14146          chr10      100189548    100189646               -  \n",
       "14147          chr10      100195029    100195171               -  \n",
       "14150          chr10      101510126    101510153               +  \n",
       "14151          chr10      101514286    101514391               +  \n",
       "14154          chr10      102010005    102010089               -  "
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Get the alternatively included exon coordinates for iPSC bimodal events\n",
    "iPSC_altexons = iPSC_bimodal_events[\"event_id\"].str.split(\"@\").str[1].str.split(\"exon:\").str[1]\n",
    "#print iPSC_altexons.head()\n",
    "iPSC_altexons_chrom = iPSC_altexons.str.split(\":\").str[0]\n",
    "iPSC_bimodal_events = iPSC_bimodal_events.assign(alt_exon_chrom=iPSC_altexons_chrom)\n",
    "#print iPSC_altexons_chrom.head()\n",
    "iPSC_altexons_start = iPSC_altexons.str.split(\":\").str[1].str.split(\"-\").str[0]\n",
    "iPSC_bimodal_events = iPSC_bimodal_events.assign(alt_exon_start=iPSC_altexons_start)\n",
    "#print iPSC_altexons_start.head()\n",
    "iPSC_altexons_end = iPSC_altexons.str.split(\":\").str[1].str.split(\"-\").str[1]\n",
    "iPSC_bimodal_events = iPSC_bimodal_events.assign(alt_exon_end=iPSC_altexons_end)\n",
    "#print iPSC_altexons_end.head()\n",
    "iPSC_altexons_strand = iPSC_altexons.str.split(\":\").str[2]\n",
    "iPSC_bimodal_events = iPSC_bimodal_events.assign(alt_exon_strand=iPSC_altexons_strand)\n",
    "#print iPSC_altexons_strand.head()\n",
    "iPSC_bimodal_events.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>phenotype</th>\n",
       "      <th>event_id</th>\n",
       "      <th>modality</th>\n",
       "      <th>alt_exon_chrom</th>\n",
       "      <th>alt_exon_start</th>\n",
       "      <th>alt_exon_end</th>\n",
       "      <th>alt_exon_strand</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>19022</th>\n",
       "      <td>iPSC</td>\n",
       "      <td>exon:chr1:100605205-100605267:+@exon:chr1:1006...</td>\n",
       "      <td>bimodal</td>\n",
       "      <td>chr1</td>\n",
       "      <td>100605738</td>\n",
       "      <td>100605807</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19031</th>\n",
       "      <td>iPSC</td>\n",
       "      <td>exon:chr1:101360189-101361554:-@exon:chr1:1013...</td>\n",
       "      <td>bimodal</td>\n",
       "      <td>chr1</td>\n",
       "      <td>101354309</td>\n",
       "      <td>101354420</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19032</th>\n",
       "      <td>iPSC</td>\n",
       "      <td>exon:chr1:101460666-101460705:-@exon:chr1:1014...</td>\n",
       "      <td>bimodal</td>\n",
       "      <td>chr1</td>\n",
       "      <td>101458193</td>\n",
       "      <td>101458296</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19033</th>\n",
       "      <td>iPSC</td>\n",
       "      <td>exon:chr1:101460666-101460705:-@exon:chr1:1014...</td>\n",
       "      <td>bimodal</td>\n",
       "      <td>chr1</td>\n",
       "      <td>101458193</td>\n",
       "      <td>101458296</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19038</th>\n",
       "      <td>iPSC</td>\n",
       "      <td>exon:chr1:101487197-101487276:-@exon:chr1:1014...</td>\n",
       "      <td>bimodal</td>\n",
       "      <td>chr1</td>\n",
       "      <td>101479266</td>\n",
       "      <td>101479374</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      phenotype                                           event_id modality  \\\n",
       "19022      iPSC  exon:chr1:100605205-100605267:+@exon:chr1:1006...  bimodal   \n",
       "19031      iPSC  exon:chr1:101360189-101361554:-@exon:chr1:1013...  bimodal   \n",
       "19032      iPSC  exon:chr1:101460666-101460705:-@exon:chr1:1014...  bimodal   \n",
       "19033      iPSC  exon:chr1:101460666-101460705:-@exon:chr1:1014...  bimodal   \n",
       "19038      iPSC  exon:chr1:101487197-101487276:-@exon:chr1:1014...  bimodal   \n",
       "\n",
       "      alt_exon_chrom alt_exon_start alt_exon_end alt_exon_strand  \n",
       "19022           chr1      100605738    100605807               +  \n",
       "19031           chr1      101354309    101354420               -  \n",
       "19032           chr1      101458193    101458296               -  \n",
       "19033           chr1      101458193    101458296               -  \n",
       "19038           chr1      101479266    101479374               -  "
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "iPSC_bimodal_events_sorted = iPSC_bimodal_events.sort_values(by=[\"alt_exon_chrom\",\"alt_exon_start\",\"alt_exon_end\"])\n",
    "iPSC_bimodal_events_sorted.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Write bed file \n",
    "iPSC_bimodal_events_sorted.iloc[:,[3,4,5,6]].to_csv(\"../processed_files/GSE85908_modalities_iPSCs_bimodalExons.bed\",sep=\"\\t\",header=False,index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GeneID</th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Strand</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>DDX11L1</td>\n",
       "      <td>chr1</td>\n",
       "      <td>11874</td>\n",
       "      <td>14409</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>WASH7P</td>\n",
       "      <td>chr1</td>\n",
       "      <td>14362</td>\n",
       "      <td>29370</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MIR6859-1</td>\n",
       "      <td>chr1</td>\n",
       "      <td>17369</td>\n",
       "      <td>17436</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MIR1302-2HG</td>\n",
       "      <td>chr1</td>\n",
       "      <td>29926</td>\n",
       "      <td>31295</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MIR1302-2</td>\n",
       "      <td>chr1</td>\n",
       "      <td>30366</td>\n",
       "      <td>30503</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        GeneID   Chr  Start    End Strand\n",
       "0      DDX11L1  chr1  11874  14409      +\n",
       "1       WASH7P  chr1  14362  29370      -\n",
       "2    MIR6859-1  chr1  17369  17436      -\n",
       "3  MIR1302-2HG  chr1  29926  31295      +\n",
       "4    MIR1302-2  chr1  30366  30503      +"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Read in genomic coordinates file to get a sense of which genes these iPSCs biomodal exons are in \n",
    "genomic_coord = pd.read_csv(\"../data/SAFformat_ExtractedGenomicCoordinatesFromGFFfile_OnlyGenes.bed\",sep=\"\\t\")\n",
    "genomic_coord.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Write to a proper bedfile for bedtools intersection\n",
    "genomic_coord.iloc[:,[1,2,3,0,4]].to_csv(\"../processed_files/GenomicCoordinatesFromGFFfile_OnlyGenes.bed\",sep=\"\\t\",header=False,index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "bedtools intersect -a ../processed_files/GSE85908_modalities_iPSCs_bimodalExons_hg38Coords.bed -b ../processed_files/GenomicCoordinatesFromGFFfile_OnlyGenes.bed -wa -wb > ../processed_files/GSE85908_modalities_iPSCs_bimodalExons_hg38Coords_Intersect_geneNames.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>100140182</td>\n",
       "      <td>100140251</td>\n",
       "      <td>+</td>\n",
       "      <td>chr1</td>\n",
       "      <td>100133080</td>\n",
       "      <td>100150402</td>\n",
       "      <td>TRMT13</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>100140182</td>\n",
       "      <td>100140251</td>\n",
       "      <td>+</td>\n",
       "      <td>chr1</td>\n",
       "      <td>100133132</td>\n",
       "      <td>100148227</td>\n",
       "      <td>TRMT13</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>100140182</td>\n",
       "      <td>100140251</td>\n",
       "      <td>+</td>\n",
       "      <td>chr1</td>\n",
       "      <td>100133132</td>\n",
       "      <td>100150496</td>\n",
       "      <td>TRMT13</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>100140182</td>\n",
       "      <td>100140251</td>\n",
       "      <td>+</td>\n",
       "      <td>chr1</td>\n",
       "      <td>100133132</td>\n",
       "      <td>100150496</td>\n",
       "      <td>TRMT13</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>100140182</td>\n",
       "      <td>100140251</td>\n",
       "      <td>+</td>\n",
       "      <td>chr1</td>\n",
       "      <td>100133132</td>\n",
       "      <td>100150496</td>\n",
       "      <td>TRMT13</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      0          1          2  3     4          5          6       7  8\n",
       "0  chr1  100140182  100140251  +  chr1  100133080  100150402  TRMT13  +\n",
       "1  chr1  100140182  100140251  +  chr1  100133132  100148227  TRMT13  +\n",
       "2  chr1  100140182  100140251  +  chr1  100133132  100150496  TRMT13  +\n",
       "3  chr1  100140182  100140251  +  chr1  100133132  100150496  TRMT13  +\n",
       "4  chr1  100140182  100140251  +  chr1  100133132  100150496  TRMT13  +"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Grab the names of the genes \n",
    "iPSC_bimodal_genes = pd.read_csv(\"../processed_files/GSE85908_modalities_iPSCs_bimodalExons_hg38Coords_Intersect_geneNames.txt\",sep=\"\\t\",header=None)\n",
    "iPSC_bimodal_genes.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>TranscriptID</th>\n",
       "      <th>Specific_IntronNumber</th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Strand</th>\n",
       "      <th>NoDMSSample_TPM</th>\n",
       "      <th>DMSTreatedSample_TPM</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ACTB</td>\n",
       "      <td>NM_001101.4</td>\n",
       "      <td>NM_001101.4_4</td>\n",
       "      <td>chr7</td>\n",
       "      <td>5529435</td>\n",
       "      <td>5529534</td>\n",
       "      <td>-</td>\n",
       "      <td>1817.102000</td>\n",
       "      <td>2949.435125</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NFKBIA</td>\n",
       "      <td>NM_020529.2</td>\n",
       "      <td>NM_020529.2_2</td>\n",
       "      <td>chr14</td>\n",
       "      <td>35402671</td>\n",
       "      <td>35402770</td>\n",
       "      <td>-</td>\n",
       "      <td>2911.698235</td>\n",
       "      <td>2152.512543</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NFKBIA</td>\n",
       "      <td>NM_020529.2</td>\n",
       "      <td>NM_020529.2_4</td>\n",
       "      <td>chr14</td>\n",
       "      <td>35403590</td>\n",
       "      <td>35403689</td>\n",
       "      <td>-</td>\n",
       "      <td>1655.983215</td>\n",
       "      <td>2057.575603</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NFKBIA</td>\n",
       "      <td>NM_020529.2</td>\n",
       "      <td>NM_020529.2_1</td>\n",
       "      <td>chr14</td>\n",
       "      <td>35402294</td>\n",
       "      <td>35402393</td>\n",
       "      <td>-</td>\n",
       "      <td>1528.822200</td>\n",
       "      <td>1764.549086</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>RPS19</td>\n",
       "      <td>NM_001321485.1</td>\n",
       "      <td>NM_001321485.1_3</td>\n",
       "      <td>chr19</td>\n",
       "      <td>41861226</td>\n",
       "      <td>41861325</td>\n",
       "      <td>+</td>\n",
       "      <td>935.644966</td>\n",
       "      <td>1759.071955</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     Gene    TranscriptID Specific_IntronNumber    Chr     Start       End  \\\n",
       "0    ACTB     NM_001101.4         NM_001101.4_4   chr7   5529435   5529534   \n",
       "1  NFKBIA     NM_020529.2         NM_020529.2_2  chr14  35402671  35402770   \n",
       "2  NFKBIA     NM_020529.2         NM_020529.2_4  chr14  35403590  35403689   \n",
       "3  NFKBIA     NM_020529.2         NM_020529.2_1  chr14  35402294  35402393   \n",
       "4   RPS19  NM_001321485.1      NM_001321485.1_3  chr19  41861226  41861325   \n",
       "\n",
       "  Strand  NoDMSSample_TPM  DMSTreatedSample_TPM  \n",
       "0      -      1817.102000           2949.435125  \n",
       "1      -      2911.698235           2152.512543  \n",
       "2      -      1655.983215           2057.575603  \n",
       "3      -      1528.822200           1764.549086  \n",
       "4      +       935.644966           1759.071955  "
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Get DMS data\n",
    "featureCounts_DMSandnonDMS = pd.read_csv(\"../processed_files/Madrasin_HiSeq_October2018_alignments/Feature_ReadCounts_100bpsInto5pIntron_SortedByTPMs.tsv\",sep=\"\\t\")\n",
    "featureCounts_DMSandnonDMS.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(210114, 9)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>TranscriptID</th>\n",
       "      <th>Specific_IntronNumber</th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Strand</th>\n",
       "      <th>NoDMSSample_TPM</th>\n",
       "      <th>DMSTreatedSample_TPM</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>RPS19</td>\n",
       "      <td>NM_001321485.1</td>\n",
       "      <td>NM_001321485.1_3</td>\n",
       "      <td>chr19</td>\n",
       "      <td>41861226</td>\n",
       "      <td>41861325</td>\n",
       "      <td>+</td>\n",
       "      <td>935.644966</td>\n",
       "      <td>1759.071955</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>RACK1</td>\n",
       "      <td>NM_006098.4</td>\n",
       "      <td>NM_006098.4_5</td>\n",
       "      <td>chr5</td>\n",
       "      <td>181241392</td>\n",
       "      <td>181241491</td>\n",
       "      <td>-</td>\n",
       "      <td>1530.989717</td>\n",
       "      <td>1150.197542</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>EIF4A2</td>\n",
       "      <td>NM_001967.3</td>\n",
       "      <td>NM_001967.3_4</td>\n",
       "      <td>chr3</td>\n",
       "      <td>186785102</td>\n",
       "      <td>186785201</td>\n",
       "      <td>+</td>\n",
       "      <td>1219.589732</td>\n",
       "      <td>834.349645</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>SEPT2</td>\n",
       "      <td>XM_024452924.1</td>\n",
       "      <td>XM_024452924.1_2</td>\n",
       "      <td>chr2</td>\n",
       "      <td>241316576</td>\n",
       "      <td>241316675</td>\n",
       "      <td>+</td>\n",
       "      <td>448.676080</td>\n",
       "      <td>806.051134</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>GSTP1</td>\n",
       "      <td>NM_000852.3</td>\n",
       "      <td>NM_000852.3_3</td>\n",
       "      <td>chr11</td>\n",
       "      <td>67584571</td>\n",
       "      <td>67584670</td>\n",
       "      <td>+</td>\n",
       "      <td>1202.972100</td>\n",
       "      <td>791.445451</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Gene    TranscriptID Specific_IntronNumber    Chr      Start        End  \\\n",
       "4    RPS19  NM_001321485.1      NM_001321485.1_3  chr19   41861226   41861325   \n",
       "16   RACK1     NM_006098.4         NM_006098.4_5   chr5  181241392  181241491   \n",
       "30  EIF4A2     NM_001967.3         NM_001967.3_4   chr3  186785102  186785201   \n",
       "33   SEPT2  XM_024452924.1      XM_024452924.1_2   chr2  241316576  241316675   \n",
       "34   GSTP1     NM_000852.3         NM_000852.3_3  chr11   67584571   67584670   \n",
       "\n",
       "   Strand  NoDMSSample_TPM  DMSTreatedSample_TPM  \n",
       "4       +       935.644966           1759.071955  \n",
       "16      -      1530.989717           1150.197542  \n",
       "30      +      1219.589732            834.349645  \n",
       "33      +       448.676080            806.051134  \n",
       "34      +      1202.972100            791.445451  "
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Overlap\n",
    "featureCounts_DMSandnonDMS_Overlap_Bimodal_iPSC = featureCounts_DMSandnonDMS[featureCounts_DMSandnonDMS[\"Gene\"].isin(iPSC_bimodal_genes[7])]\n",
    "print featureCounts_DMSandnonDMS_Overlap_Bimodal_iPSC.shape\n",
    "featureCounts_DMSandnonDMS_Overlap_Bimodal_iPSC.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(311, 9)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>TranscriptID</th>\n",
       "      <th>Specific_IntronNumber</th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Strand</th>\n",
       "      <th>NoDMSSample_TPM</th>\n",
       "      <th>DMSTreatedSample_TPM</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>RPS19</td>\n",
       "      <td>NM_001321485.1</td>\n",
       "      <td>NM_001321485.1_3</td>\n",
       "      <td>chr19</td>\n",
       "      <td>41861226</td>\n",
       "      <td>41861325</td>\n",
       "      <td>+</td>\n",
       "      <td>935.644966</td>\n",
       "      <td>1759.071955</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>RACK1</td>\n",
       "      <td>NM_006098.4</td>\n",
       "      <td>NM_006098.4_5</td>\n",
       "      <td>chr5</td>\n",
       "      <td>181241392</td>\n",
       "      <td>181241491</td>\n",
       "      <td>-</td>\n",
       "      <td>1530.989717</td>\n",
       "      <td>1150.197542</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>EIF4A2</td>\n",
       "      <td>NM_001967.3</td>\n",
       "      <td>NM_001967.3_4</td>\n",
       "      <td>chr3</td>\n",
       "      <td>186785102</td>\n",
       "      <td>186785201</td>\n",
       "      <td>+</td>\n",
       "      <td>1219.589732</td>\n",
       "      <td>834.349645</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>SEPT2</td>\n",
       "      <td>XM_024452924.1</td>\n",
       "      <td>XM_024452924.1_2</td>\n",
       "      <td>chr2</td>\n",
       "      <td>241316576</td>\n",
       "      <td>241316675</td>\n",
       "      <td>+</td>\n",
       "      <td>448.676080</td>\n",
       "      <td>806.051134</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>GSTP1</td>\n",
       "      <td>NM_000852.3</td>\n",
       "      <td>NM_000852.3_3</td>\n",
       "      <td>chr11</td>\n",
       "      <td>67584571</td>\n",
       "      <td>67584670</td>\n",
       "      <td>+</td>\n",
       "      <td>1202.972100</td>\n",
       "      <td>791.445451</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>GSTP1</td>\n",
       "      <td>NM_000852.3</td>\n",
       "      <td>NM_000852.3_5</td>\n",
       "      <td>chr11</td>\n",
       "      <td>67585242</td>\n",
       "      <td>67585341</td>\n",
       "      <td>+</td>\n",
       "      <td>899.519678</td>\n",
       "      <td>756.756954</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>RPS5</td>\n",
       "      <td>NM_001009.3</td>\n",
       "      <td>NM_001009.3_4</td>\n",
       "      <td>chr19</td>\n",
       "      <td>58393488</td>\n",
       "      <td>58393587</td>\n",
       "      <td>+</td>\n",
       "      <td>519.481645</td>\n",
       "      <td>705.637063</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>GCN1</td>\n",
       "      <td>NM_006836.1</td>\n",
       "      <td>NM_006836.1_36</td>\n",
       "      <td>chr12</td>\n",
       "      <td>120161390</td>\n",
       "      <td>120161489</td>\n",
       "      <td>-</td>\n",
       "      <td>294.782352</td>\n",
       "      <td>702.898498</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>SF3B3</td>\n",
       "      <td>NM_012426.4</td>\n",
       "      <td>NM_012426.4_14</td>\n",
       "      <td>chr16</td>\n",
       "      <td>70556335</td>\n",
       "      <td>70556434</td>\n",
       "      <td>+</td>\n",
       "      <td>406.770746</td>\n",
       "      <td>700.159932</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>CHD2</td>\n",
       "      <td>NM_001271.3</td>\n",
       "      <td>NM_001271.3_26</td>\n",
       "      <td>chr15</td>\n",
       "      <td>92985674</td>\n",
       "      <td>92985773</td>\n",
       "      <td>+</td>\n",
       "      <td>330.907641</td>\n",
       "      <td>625.305806</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Gene    TranscriptID Specific_IntronNumber    Chr      Start        End  \\\n",
       "4    RPS19  NM_001321485.1      NM_001321485.1_3  chr19   41861226   41861325   \n",
       "16   RACK1     NM_006098.4         NM_006098.4_5   chr5  181241392  181241491   \n",
       "30  EIF4A2     NM_001967.3         NM_001967.3_4   chr3  186785102  186785201   \n",
       "33   SEPT2  XM_024452924.1      XM_024452924.1_2   chr2  241316576  241316675   \n",
       "34   GSTP1     NM_000852.3         NM_000852.3_3  chr11   67584571   67584670   \n",
       "38   GSTP1     NM_000852.3         NM_000852.3_5  chr11   67585242   67585341   \n",
       "44    RPS5     NM_001009.3         NM_001009.3_4  chr19   58393488   58393587   \n",
       "45    GCN1     NM_006836.1        NM_006836.1_36  chr12  120161390  120161489   \n",
       "46   SF3B3     NM_012426.4        NM_012426.4_14  chr16   70556335   70556434   \n",
       "60    CHD2     NM_001271.3        NM_001271.3_26  chr15   92985674   92985773   \n",
       "\n",
       "   Strand  NoDMSSample_TPM  DMSTreatedSample_TPM  \n",
       "4       +       935.644966           1759.071955  \n",
       "16      -      1530.989717           1150.197542  \n",
       "30      +      1219.589732            834.349645  \n",
       "33      +       448.676080            806.051134  \n",
       "34      +      1202.972100            791.445451  \n",
       "38      +       899.519678            756.756954  \n",
       "44      +       519.481645            705.637063  \n",
       "45      -       294.782352            702.898498  \n",
       "46      +       406.770746            700.159932  \n",
       "60      +       330.907641            625.305806  "
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#TPM greater 100\n",
    "featureCounts_DMSandnonDMS_Overlap_Bimodal_iPSC_GreaterThan100 = featureCounts_DMSandnonDMS_Overlap_Bimodal_iPSC[featureCounts_DMSandnonDMS_Overlap_Bimodal_iPSC[\"DMSTreatedSample_TPM\"]>100]\n",
    "print featureCounts_DMSandnonDMS_Overlap_Bimodal_iPSC_GreaterThan100.shape\n",
    "featureCounts_DMSandnonDMS_Overlap_Bimodal_iPSC_GreaterThan100.head(n=10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.15rc1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
