#!/bin/bash

fastqfolder=

#Get paired reads
#echo "Getting paired reads DMS..."
samtools view ../DMSTreatedSample_TAGGCATG_S1.bam | fgrep -w -f DMSTreatedSample_TAGGCATG_S1_AlignedToGSTP1_MAPQgreaterThan10_SortedByCoords_MaturemRNA_Sorted_ReadNames.txt > DMSTreatedSample_TAGGCATG_S1_AlignedToGSTP1_MAPQgreaterThan10_SortedByCoords_MaturemRNA_Sorted_PairedReads.sam
samtools view ../DMSTreatedSample_TAGGCATG_S1.bam | fgrep -w -f DMSTreatedSample_TAGGCATG_S1_AlignedToGSTP1_MAPQgreaterThan10_SortedByCoords_PremRNA_Sorted_ReadNames.txt > DMSTreatedSample_TAGGCATG_S1_AlignedToGSTP1_MAPQgreaterThan10_SortedByCoords_PremRNA_Sorted_PairedReads.sam

#Concatenate paired reads sam with header
echo "Concatenating header DMS..."
cat ../DMSTreatedSample_TAGGCATG_S1_Header.sam DMSTreatedSample_TAGGCATG_S1_AlignedToGSTP1_MAPQgreaterThan10_SortedByCoords_MaturemRNA_Sorted_PairedReads.sam > DMSTreatedSample_TAGGCATG_S1_AlignedToGSTP1_MAPQgreaterThan10_SortedByCoords_MaturemRNA_Sorted_PairedReads_PlusHeader.sam
cat ../DMSTreatedSample_TAGGCATG_S1_Header.sam DMSTreatedSample_TAGGCATG_S1_AlignedToGSTP1_MAPQgreaterThan10_SortedByCoords_PremRNA_Sorted_PairedReads.sam > DMSTreatedSample_TAGGCATG_S1_AlignedToGSTP1_MAPQgreaterThan10_SortedByCoords_PremRNA_Sorted_PairedReads_PlusHeader.sam

#Convert sam file to bam file
echo "Convert samfile to bam and sort DMS..."
samtools view -u DMSTreatedSample_TAGGCATG_S1_AlignedToGSTP1_MAPQgreaterThan10_SortedByCoords_MaturemRNA_Sorted_PairedReads_PlusHeader.sam | samtools sort -o DMSTreatedSample_TAGGCATG_S1_AlignedToGSTP1_MAPQgreaterThan10_SortedByCoords_MaturemRNA_Sorted_PairedReads_Sorted.bam
samtools view -u DMSTreatedSample_TAGGCATG_S1_AlignedToGSTP1_MAPQgreaterThan10_SortedByCoords_PremRNA_Sorted_PairedReads_PlusHeader.sam | samtools sort -o DMSTreatedSample_TAGGCATG_S1_AlignedToGSTP1_MAPQgreaterThan10_SortedByCoords_PremRNA_Sorted_PairedReads_Sorted.bam
    
#Convert bam to fastq 
echo "Converting to fastq DMS..."
samtools fastq -N -1 DMSTreatedSample_TAGGCATG_S1_AlignedToGSTP1_MAPQgreaterThan10_SortedByCoords_MaturemRNA_Sorted_R1.fastq -2 DMSTreatedSample_TAGGCATG_S1_AlignedToGSTP1_MAPQgreaterThan10_SortedByCoords_MaturemRNA_Sorted_R2.fastq DMSTreatedSample_TAGGCATG_S1_AlignedToGSTP1_MAPQgreaterThan10_SortedByCoords_MaturemRNA_Sorted_PairedReads_Sorted.bam
samtools fastq -N -1 DMSTreatedSample_TAGGCATG_S1_AlignedToGSTP1_MAPQgreaterThan10_SortedByCoords_PremRNA_Sorted_R1.fastq -2 DMSTreatedSample_TAGGCATG_S1_AlignedToGSTP1_MAPQgreaterThan10_SortedByCoords_PremRNA_Sorted_R2.fastq DMSTreatedSample_TAGGCATG_S1_AlignedToGSTP1_MAPQgreaterThan10_SortedByCoords_PremRNA_Sorted_PairedReads_Sorted.bam

#Get paired reads
echo "Getting paired reads noDMS..."
samtools view ../NoDMSSample_CTCTCTAC_S2.bam | fgrep -w -f NoDMSSample_CTCTCTAC_S2_AlignedToGSTP1_MAPQgreaterThan10_SortedByCoords_MaturemRNA_Sorted_ReadNames.txt > NoDMSSample_CTCTCTAC_S2_AlignedToGSTP1_MAPQgreaterThan10_SortedByCoords_MaturemRNA_Sorted_PairedReads.sam
samtools view ../NoDMSSample_CTCTCTAC_S2.bam | fgrep -w -f NoDMSSample_CTCTCTAC_S2_AlignedToGSTP1_MAPQgreaterThan10_SortedByCoords_PremRNA_Sorted_ReadNames.txt > NoDMSSample_CTCTCTAC_S2_AlignedToGSTP1_MAPQgreaterThan10_SortedByCoords_PremRNA_Sorted_PairedReads.sam

#Concatenate paired reads sam with header
echo "Concatenating header noDMS..."
cat ../NoDMSSample_CTCTCTAC_S2_Header.sam NoDMSSample_CTCTCTAC_S2_AlignedToGSTP1_MAPQgreaterThan10_SortedByCoords_MaturemRNA_Sorted_PairedReads.sam > NoDMSSample_CTCTCTAC_S2_AlignedToGSTP1_MAPQgreaterThan10_SortedByCoords_MaturemRNA_Sorted_PairedReads_PlusHeader.sam
cat ../NoDMSSample_CTCTCTAC_S2_Header.sam NoDMSSample_CTCTCTAC_S2_AlignedToGSTP1_MAPQgreaterThan10_SortedByCoords_PremRNA_Sorted_PairedReads.sam > NoDMSSample_CTCTCTAC_S2_AlignedToGSTP1_MAPQgreaterThan10_SortedByCoords_PremRNA_Sorted_PairedReads_PlusHeader.sam

#Convert sam file to bam file
echo "Convert samfile to bam and sort noDMS..."
samtools view -u NoDMSSample_CTCTCTAC_S2_AlignedToGSTP1_MAPQgreaterThan10_SortedByCoords_MaturemRNA_Sorted_PairedReads_PlusHeader.sam | samtools sort -o NoDMSSample_CTCTCTAC_S2_AlignedToGSTP1_MAPQgreaterThan10_SortedByCoords_MaturemRNA_Sorted_PairedReads_Sorted.bam
samtools view -u NoDMSSample_CTCTCTAC_S2_AlignedToGSTP1_MAPQgreaterThan10_SortedByCoords_PremRNA_Sorted_PairedReads_PlusHeader.sam | samtools sort -o NoDMSSample_CTCTCTAC_S2_AlignedToGSTP1_MAPQgreaterThan10_SortedByCoords_PremRNA_Sorted_PairedReads_Sorted.bam

#Convert bam to fastq
echo "Converting to fastq noDMS..."
samtools fastq -N -1 NoDMSSample_CTCTCTAC_S2_AlignedToGSTP1_MAPQgreaterThan10_SortedByCoords_MaturemRNA_Sorted_R1.fastq -2 NoDMSSample_CTCTCTAC_S2_AlignedToGSTP1_MAPQgreaterThan10_SortedByCoords_MaturemRNA_Sorted_R2.fastq NoDMSSample_CTCTCTAC_S2_AlignedToGSTP1_MAPQgreaterThan10_SortedByCoords_MaturemRNA_Sorted_PairedReads_Sorted.bam
samtools fastq -N -1 NoDMSSample_CTCTCTAC_S2_AlignedToGSTP1_MAPQgreaterThan10_SortedByCoords_PremRNA_Sorted_R1.fastq -2 NoDMSSample_CTCTCTAC_S2_AlignedToGSTP1_MAPQgreaterThan10_SortedByCoords_PremRNA_Sorted_R2.fastq NoDMSSample_CTCTCTAC_S2_AlignedToGSTP1_MAPQgreaterThan10_SortedByCoords_PremRNA_Sorted_PairedReads_Sorted.bam